data-prepper-plugins/mutate-string-processors/src/test/java/org/opensearch/dataprepper/plugins/processor/mutatestring/TruncateStringProcessorTests.java

```diff
@@ -0,0 +1,115 @@
+    }
+
+    @ParameterizedTest
+    @ArgumentsSource(TruncateStringArgumentsProvider.class)
+    void testTruncateStringProcessor(final String message, final int truncateLength, final String truncatedMessage) {
+
+        when(config.getIterativeConfig()).thenReturn(Collections.singletonList(createEntry("message", truncateLength, null)));
+
+        final TruncateStringProcessor truncateStringProcessor = createObjectUnderTest();
+        final Record<Event> record = createEvent(message);
+        final List<Record<Event>> truncatedRecords = (List<Record<Event>>) truncateStringProcessor.doExecute(Collections.singletonList(record));
+        assertThat(truncatedRecords.get(0).getData().get("message", Object.class), notNullValue());
+        assertThat(truncatedRecords.get(0).getData().get("message", Object.class), equalTo(truncatedMessage));
+    }
+
+    public void testLengthNotDefinedThrowsError() {
+        when(config.getIterativeConfig()).thenReturn(Collections.singletonList(createEntry("message", null, null)));
+        when(config.getEntries()).thenReturn(Collections.singletonList(createEntry("message", null, null)));
+
+        assertThrows(IllegalArgumentException.class, () -> createObjectUnderTest());
+    }
+
+    @Test
+    void test_event_is_the_same_when_truncateWhen_condition_returns_false() {
+        final String truncateWhen = UUID.randomUUID().toString();
+        final String message = UUID.randomUUID().toString();
+
+        when(config.getIterativeConfig()).thenReturn(Collections.singletonList(createEntry("message", 5, truncateWhen)));
+
+        final TruncateStringProcessor truncateStringProcessor = createObjectUnderTest();
+        final Record<Event> record = createEvent(message);
+        when(expressionEvaluator.evaluateConditional(truncateWhen, record.getData())).thenReturn(false);
+        final List<Record<Event>> truncatedRecords = (List<Record<Event>>) truncateStringProcessor.doExecute(Collections.singletonList(record));
+
+        assertThat(truncatedRecords.get(0).getData().toMap(), equalTo(record.getData().toMap()));
+    }
+
+
+    private TruncateStringProcessorConfig.Entry createEntry(final String source, final Integer length, final String truncateWhen) {
+        return new TruncateStringProcessorConfig.Entry(source, length, truncateWhen);
+    }
+
+    private Record<Event> createEvent(final String message) {
+        final Map<String, Object> eventData = new HashMap<>();
+        eventData.put("message", message);
+        return new Record<>(JacksonEvent.builder()
+                .withEventType("event")
+                .withData(eventData)
+                .build());
+    }
+
+    static class TruncateStringArgumentsProvider implements ArgumentsProvider {
+
+        @Override
+        public Stream<? extends Arguments> provideArguments(ExtensionContext context) {
+            return Stream.of(
+                    Arguments.arguments("hello,world,no-truncate", 100, "hello,world,no-truncate"),
```

Comment:
I think it would be nice to add a static import on `Arguments.arguments` such that these lines just look like:

```
arguments("hello,world,no-truncate", 100, "hello,world,no-truncate")
```